content/geoip/geolocate-an-ip/databases.mdx

```diff
@@ -1,18 +1,249 @@
 ---
 draft: false
-title: (todo)
+title: Geolocate an IP address using MaxMind Databases
 ---
 
-(todo)
+Geolocating an IP address using MaxMind databases consists of configuring a
+database reader and querying the database.
 
-## Official API Clients
+## Implementation
 
-(todo)
+MaxMind offers and highly recommends using official client libraries to query
+our databases.
+
+### 1. Install the GeoIP2 client library
+
+We have a collection of officially supported libraries for you to query with
+the GeoIP2 and GeoLite2 databases:
+
+<CodeSet>
+
+```cli-csharp
+# Install via NuGet
+Install-Package MaxMind.GeoIP2
+```
+
+```cli-java
+# Install via Maven, recommended
+<dependency>
+  <groupId>com.maxmind.geoip2</groupId>
+  <artifactId>geoip2</artifactId>
+  <version>2.15.0</version>
+</dependency>
+
+# Or install via Gradle
+repositories {
+  mavenCentral()
+}
+dependencies {
+  compile 'com.maxmind.geoip2:geoip2:2.15.0'
+}
+```
+
+```cli-javascript
+# Install via npm
+npm install @maxmind/geoip2-node
+
+# Or install via yarn
+yarn add @maxmind/geoip2-node
+```
+
+```cli-php
+# Install via Composer
+composer require geoip2/geoip2:~2.0
+```
+
+```cli-python
+# Install via pip
+pip install geoip2 
+```
+
+```cli-ruby
+# Install as a gem
+gem install maxmind-geoip2 
+
+# Or add this to your Gemfile
+gem 'maxmind-geoip2'
+```
+
+</CodeSet>
+
+### 2. Configure a GeoIP2 database reader to query the database
+
+Configuring the database reader requires the database file to be accessible on
+the filesystem.  After configuring the database reader, you can then query the
+database by calling the method corresponding to the database type (e.g. `city`
+or `country`), passing it the IP address you want to lookup.
```

Comment:
I understand what this is trying to say, but I think that it would be clearer to swap `, passing it` with `and passing it`.

Also we want `look up` (verb) rather than `lookup` (noun) here.